checkov/cloudformation/checks/resource/aws/AmazonMQBrokerPublicAccess.py

```diff
@@ -0,0 +1,28 @@
+from checkov.common.models.enums import CheckResult, CheckCategories
+from checkov.cloudformation.checks.resource.base_resource_check import BaseResourceCheck
+
+class AmazonMQBrokerPublicAccess(BaseResourceCheck):
+
+    def __init__(self):
+        name = "Ensure Amazon MQ Broker should not have public access"
+        id = "CKV_AWS_166"
+        supported_resources = ['AWS::AmazonMQ::Broker']
+        categories = [CheckCategories.GENERAL_SECURITY]
+        super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
+
+    def scan_resource_conf(self, conf):
```

Comment:
I think in general you can make the check much simpler, by subclassing `BaseResourceValueCheck` like here
https://github.com/bridgecrewio/checkov/blob/1e8a4aa2bae94bf571fe73f88338075f91056d59/checkov/cloudformation/checks/resource/aws/RDSPubliclyAccessible.py#L5-L19

Reply:
@gruebel  made the changes in code as per your suggestions.